src/briefcase/console.py

```diff
@@ -118,6 +162,67 @@ def error(self, message="", *, prefix="", markup=False):
         """Log message at error level; always included in output."""
         self._log(prefix=prefix, message=message, markup=markup, style="bold red")
 
+    def capture_stacktrace(self):
+        """Extract Rich stacktrace from exception while in except block."""
+        self.stacktrace = Traceback.extract(*sys.exc_info(), show_locals=True)
+
+    def save_log_to_file(self, command):
+        """Save the current application log to file."""
+        # only save log if a command ran and it errored or --save-log passed
+        if command is None or (not self.stacktrace and not command.save_log):
+            return
+
+        with command.input.wait_bar("Saving application log...", transient=True):
+            self.cleanup_log_files(command.logs_path)
+
+            if self.stacktrace:
+                self.rich_log_file_console.print()
+                self.rich_log_file_console.print(
+                    Traceback(
+                        trace=self.stacktrace,
+                        width=LOG_FILE_WIDTH,
+                        show_locals=True,
+                    )
+                )
+
+            now = datetime.now()
+            log_filepath = (
+                command.logs_path / f"briefcase_{now.strftime('%Y_%m_%d-%H_%M_%S')}.log"
+            )
+            rich_console.print()
+            try:
+                command.logs_path.mkdir(parents=True, exist_ok=True)
+                with open(
+                    log_filepath, mode="w", encoding="utf-8", errors="ignore"
+                ) as log_file:
+                    log_file.write(
+                        f"Date: {now.strftime('%Y-%m-%d')}\n"
+                        f"Command line: {' '.join(sys.argv)}\n"
+                        f"Briefcase version: {__version__}\n"
+                        f"Platform: {sys.platform}\n"
+                        f"Target platform: {command.platform}\n"
+                        f"Target format: {command.output_format}\n"
+                        "\n"
+                        "Application Log:\n"
+                        f"{self.rich_log_file_console.export_text()}"
+                    )
+            except OSError as exc:
+                self.error(f"Failed to save log to {log_filepath}: {exc}")
+            else:
+                self.info(f"Wrote application log to {log_filepath}")
+
+    def cleanup_log_files(self, logs_path):
+        """Delete old log files."""
+        maximum_log_file_age = 14  # days
+        age_limit = datetime.now().timestamp() - (maximum_log_file_age * 24 * 60 * 60)
+        log_filenames = next(os.walk(logs_path), (None, None, []))[2]
+        for log_file in (logs_path / file for file in log_filenames):
+            if log_file.stat().st_mtime < age_limit:
+                try:
+                    log_file.unlink()
+                except OSError:
+                    pass
```

Comment:
Wasn't sure if it was worth logging that a previously created log file couldn't be deleted....it seems like a strange situation...alternatively, we could nag the user to delete them themselves...

Reply:
Closing since users will be expected to manage log files making this moot.